sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3WritableByteChannel.java

```diff
@@ -118,13 +118,22 @@ public int write(ByteBuffer sourceBuffer) throws IOException {
 
     int totalBytesWritten = 0;
     while (sourceBuffer.hasRemaining()) {
+      int position = sourceBuffer.position();
       int bytesWritten = Math.min(sourceBuffer.remaining(), uploadBuffer.remaining());
       totalBytesWritten += bytesWritten;
 
-      byte[] copyBuffer = new byte[bytesWritten];
-      sourceBuffer.get(copyBuffer);
-      uploadBuffer.put(copyBuffer);
-      md5.update(copyBuffer);
+      if (sourceBuffer.hasArray()) {
```

Comment:
Could you add a Javadoc for this method about this to make it more evident why it's needed?

Reply:
@aromanenko-dev I added some comments to explain. Is that sufficient?